### PR TITLE
Add `Alerts` extra

### PR DESCRIPTION
**Pull request by mhils**
This PR adds an `Alert` extra, which can be used to render [GitHub Markdown Alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).

### CODE REVIEW

File: lib/markdown2.py [hunk 1]
```diff
@@ -2722,6 +2722,41 @@ def run(self, text):
...
+        </blockquote>
+    ''', re.X
+    )
+
+    def test(self, text):
+        return "<blockquote>" in text
+
+    def sub(self, match: re.Match) -> str:
+        typ = match["type"].lower()
+        heading = f"<em>{match["type"].title()}</em>"
```
Comment (nicholasserra) [inline lib/markdown2.py:2749]: Bad f string here? CI failing on it

**Comment by nicholasserra** [comment]
Thanks! Will review fully asap

File: lib/markdown2.py (hunk 1 above)
Comment (mhils) [reply lib/markdown2.py]: Thanks! I should have tested not only on 3.12 where this works. :)

File: test/tm-cases/alerts.text [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+> Helpful advice for doing things better or more easily.
+
+> [!IMPORTANT]
+> Key information users need to know to achieve their goal.
+
+> [!WARNING]
+> Urgent info that needs immediate user attention to avoid problems.
+
+> [!CAUTION]
+>
```
Comment (nicholasserra) [inline test/tm-cases/alerts.text:14]: Is the extra line needed here? Wasn't sure if this was to catch an edge case or not

**Comment by nicholasserra** [comment]
LGTM thank you!
